VideoWeb/VideoWeb.ChatHub/Hub/ChatHub.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging;
+using VideoWeb.Common.SignalR;
+using VideoWeb.Services.Video;
+
+namespace VideoWeb.ChatHub.Hub
+{
+    public interface IChatHubClient
+    {
+        Task ReceiveMessage(Guid conferenceId, string from, string message, DateTime timestamp);
+    }
+
+    [Authorize(Policy = "EventHubUser")]
+    public class ChatHub : Hub<IChatHubClient>
+    {
+        private readonly IUserProfileService _userProfileService;
+        private readonly IVideoApiClient _videoApiClient;
+
+        private readonly ILogger<ChatHub> _logger;
+
+        public ChatHub(IUserProfileService userProfileService, IVideoApiClient videoApiClient, ILogger<ChatHub> logger)
+        {
+            _userProfileService = userProfileService;
+            _videoApiClient = videoApiClient;
+            _logger = logger;
+        }
+        public override async Task OnConnectedAsync()
+        {
+            var userName = await GetObfuscatedUsernameAsync(Context.User.Identity.Name);
+            _logger.LogTrace($"Connected to chat hub server-side: { userName } ");
+            var conferences = await GetConferencesForUser(Context.User.Identity.Name);
+            AddToConferences(conferences);
+
+            await base.OnConnectedAsync();
+        }
+
+        public override async Task OnDisconnectedAsync(Exception exception)
+        {
+            var userName = await GetObfuscatedUsernameAsync(Context.User.Identity.Name);
+            _logger.LogCritical(exception, $"Disconnected from chat hub server-side: { userName } ");
```

Comment:
Is this really a critical as if it disconnects normally e.g. on log out, we dont want to log as critical but as warning ??

Reply:
I'll monitor this a bit more... i want to track if an exception is thrown every time on say a page refresh and when i sleep or kill the connection

Reply:
the exception property is null until there's a genuine issue. if the exception is null i've logged as info rather than critical